cocos/asset/assets/simple-texture.ts

```diff
@@ -155,32 +178,24 @@ export class SimpleTexture extends TextureBase {
         region.texExtent.height = this._textureHeight >> level;
         region.texSubres.mipLevel = level;
         region.texSubres.baseArrayLayer = arrayIndex;
-
-        if (DEV) {
-            if (source instanceof HTMLElement) {
-                if (source.height > region.texExtent.height
-                    || source.width > region.texExtent.width) {
-                    error(`Image source(${this.name}) bounds override.`);
-                }
-            }
-        }
-
-        if (ArrayBuffer.isView(source)) {
-            gfxDevice.copyBuffersToTexture([source], this._gfxTexture, _regions);
+        let imageData;
+        if (source instanceof ImageData) {
+            imageData = source;
         } else {
-            gfxDevice.copyTexImagesToTexture([source], this._gfxTexture, _regions);
+            // This is a hack method, otherwise ts will just report an error.
+            imageData = new ImageData(source as IMemoryImageSource);
         }
+        gfxDevice.copyImageDatasToTexture([imageData], this._gfxTexture, _regions);
     }
 
     /**
      * @engineInternal
      */
     protected _assignImage (image: ImageAsset, level: number, arrayIndex?: number): void {
-        const data = image.data;
-        if (!data) {
+        if (!image.data) {
```

Comment:
Should it be `if (!image.imageData) {` ?

Reply:
No need, the data interface maintains compatibility and can be used in this way